HACKING.rst

```diff
@@ -34,28 +34,40 @@ version by running::
 All that separates now from the fully functioning development environment is a
 few dependencies. We can install them by running the next command::
 
-   (venv) $ pip install \
-              -r sanity.requirements \
-              -r units.requirements \
-              -r integration.requirements
+   (venv) $ make requirements
 
 And this is it. We are all set now. To validate our setup, we can run the
 bundled tests::
 
+   (venv) $ make sanity
+   (venv) $ make units
+   (venv) $ make -j8 integration
+
+We may want to go grab a cup of tea after running that last command, since it
+will take a while (about ten minutes on a semi-decent development machine).
+
+Also, for that last batch of tests, we need to have docker installed,
+configured, and running. And that is it. If nothing broke, we are ready to
+start developing.
+
+
+Section with content that will be reworked into a development story
+-------------------------------------------------------------------
+
+TODO: Describe the process of adding a new module.
+
    (venv) $ ansible-test sanity --python 3.7
    (venv) $ ansible-test units --python 3.7
    (venv) $ export ANSIBLE_COLLECTIONS_PATHS=$(pwd)/../../..
-   (venv) $ cd tests/integration/modules
-   (venv) $ molecule --base-config molecule/shared/base.yml test --all
+   (venv) $ cd tests/integration
+   (venv) $ molecule --base-config base.yml test -s module_asset
 
 .. note::
```

Comment:
If we update L59 and L60 so that they use `make` with new targets, this warning is not needed anymore as the Makefile takes care of determining python version for us.

Reply:
Would it be OK with you if we keep this in for now? I promise I will move this into a more appropriate document under the `docs` directory as soon as possible.

Why would I like to keep this? While I really like the wrappers, I also feel that it is still immensely valuable to know how things work under the hood. Once things start to go south, knowing a thing or two about the internal workings can be a lifesaver. And yes, I know I can always dig this out of the git's history, I would instead move this content in a single commit.

Reply:
Sure, I understand the concern and can't say I disagree, so :+1: . I brought this up more for the sake of consistency with the first part of the updated `HACKING.rst` document which now includes instructions with `make`. 

Still, once we follow up with the dedicated document under `docs`, let's not forget to at least mention the possibility of using convenience wrappers as an alternative to running commands stated here :wink:

Reply:
What I had in mind is to keep convenience wrappers in a quickstart and other casual documents that deal with tests on a higher level. Low level commands are destined for a document about trouble shooting the testing process.